src/ray/raylet/node_manager.cc

```diff
@@ -397,6 +399,16 @@ void NodeManager::ClientRemoved(const ClientTableDataT &client_data) {
 
   // Remove the remote server connection.
   remote_server_connections_.erase(client_id);
+
+  // For any live actors that were on the dead node, broadcast a notification
+  // about the actor's death
+  // TODO(swang): This could be very slow if there are many actors.
+  for (const auto &actor_entry : actor_registry_) {
```

Comment:
We never remove anything from `actor_registry_`, right? So if many jobs create many actors over the course of the cluster's lifetime, this will become an issue.

Maybe `HandleDisconnectedActor` is the right time to GC stuff? Also, `HandleDriverTableUpdate` probably.

Reply:
The GC problem is the same as before and out of scope of this PR.